src/org/omegat/core/machinetranslators/ApertiumTranslate.java

```diff
@@ -135,31 +137,23 @@ protected String translate(Language sLang, Language tLang, String text) throws E
 
     @SuppressWarnings("unchecked")
     protected String getJsonResults(String json) {
-        Map<String, Object> rootNode;
+        JsonNode rootNode;
+        ObjectMapper mapper = new ObjectMapper();
         try {
-            rootNode = (Map<String, Object>) JsonParser.parse(json);
+            rootNode = mapper.readTree(json);
         } catch (Exception e) {
             Log.logErrorRB(e, "MT_JSON_ERROR");
             return OStrings.getString("MT_JSON_ERROR");
         }
 
-        Integer code = 0;
-        String tr = null;
-        if (rootNode.containsKey("responseStatus")) {
-            code = (Integer) rootNode.get("responseStatus");
-        } else {
-            return OStrings.getString("APERTIUM_CUSTOM_SERVER_INVALID");
-        }
-
-        if (rootNode.containsKey("responseData")) {
-            Map<String, Object> data = (Map<String, Object>) rootNode.get("responseData");
-            tr = (String) data.get("translatedText");
-        }
+        Integer code = rootNode.get("responseStatus").asInt();
+        // return OStrings.getString("APERTIUM_CUSTOM_SERVER_INVALID");
```

Comment:
The old code checked that the `responseStatus` key was present. The new code assumes it is.

I don't know enough about Apertium to know if this is a safe assumption to make.

If it's OK to assume this, then we should remove the code. If not, then we should check for the key.

Reply:
I've added test code and improve the check.